blocnotes/blocnotes/MasterViewController.m

```diff
@@ -62,14 +62,45 @@ - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
 {
     if ([[segue identifier] isEqualToString:@"showDetail"])
     {
-        NSIndexPath *indexPath = [self.tableView indexPathForSelectedRow];
-        NSManagedObject *object = [[self fetchedResultsController] objectAtIndexPath:indexPath];
-        DetailViewController *controller = (DetailViewController *)[[segue destinationViewController] topViewController];
-        [controller setDelegate:self];
-        [controller setDetailItem:object];
-        controller.navigationItem.leftBarButtonItem = self.splitViewController.displayModeButtonItem;
-        controller.navigationItem.leftItemsSupplementBackButton = YES;
+        if (self.isNew == YES)
+        {
+            NSManagedObjectContext *context = [self.fetchedResultsController managedObjectContext];
+            NSEntityDescription *entity = [[self.fetchedResultsController fetchRequest] entity];
+            NSManagedObject *newManagedObject = [NSEntityDescription insertNewObjectForEntityForName:[entity name] inManagedObjectContext:context];
+            
+            DetailViewController *controller = (DetailViewController *)[[segue destinationViewController] topViewController];
+            [controller setDetailItem:newManagedObject];
+            controller.navigationItem.leftBarButtonItem = self.splitViewController.displayModeButtonItem;
+            controller.navigationItem.leftItemsSupplementBackButton = YES;
+        }
+        
+        else
+        {
+            NSIndexPath *indexPath = [self.tableView indexPathForSelectedRow];
+            NSManagedObject *object = nil;
+            
+            if (self.searchController.isActive)
+            {
+                object = [self.filteredList objectAtIndex:indexPath.row];
+            }
+            
+            else
+            {
+                object = [[self fetchedResultsController] objectAtIndexPath:indexPath];
+            }
+            
+            DetailViewController *controller = (DetailViewController *)[[segue destinationViewController] topViewController];
+            [controller setDetailItem:object];
+            controller.navigationItem.leftBarButtonItem = self.splitViewController.displayModeButtonItem;
+            controller.navigationItem.leftItemsSupplementBackButton = YES;
+            [self.searchController dismissViewControllerAnimated:YES completion:nil];
```

Comment:
I would strongly consider getting rid of these next three lines:
```
            [self.searchController dismissViewControllerAnimated:YES completion:nil];
            self.searchController.searchBar.text = @"";
            [self.tableView reloadData];
```
The person may need to muck around in the search result a bunch to find the one they really want, so I would, as a user, expect to be returned to my search results when I back out of my note.